core/src/main/java/org/mule/runtime/core/api/Event.java

```diff
@@ -48,73 +47,20 @@
   /**
    * @return the context applicable to all events created from the same root {@link Event} from a {@link MessageSource}.
    */
-  EventContext getContext();
+  EventContext getInternalContext();
 
   /**
-   * Returns the correlation metadata of this message. See {@link GroupCorrelation}.
+   * Internal parameters used by the runtime to pass information around.
    * 
-   * @return the correlation metadata of this message.
-   */
-  Optional<GroupCorrelation> getGroupCorrelation();
-
-  /**
-   * The returned value will depend on the {@link MessageSource} that created this event, and the flow that is executing the
-   * event.
-   *
-   * @return the correlation id to use for this event.
-   */
-  String getCorrelationId();
-
-  /**
-   * Returns the variable registered under the given {@code key}
-   *
-   * @param key the name or key of the variable. This must be non-null.
-   * @param <T> the type of the variable value.
-   * @return a {@link TypedValue} containing the variable's value and {@link DataType}
-   * @throws java.util.NoSuchElementException if the flow variable does not exist.
    */
-  <T> TypedValue<T> getVariable(String key);
+  Map<String, ?> getInternalParameters();
```

Comment:
Isn't it best to start using `TypedValue` more consistently in core rather than introducing a second set of API methods to support `Object` instead of `TypedValue`? Or did i misunderstand this and they really are two different distinct sets of parameters?